### PR TITLE
video listing and video detail page

### PR DESCRIPTION
**Pull request by praveenreddy1798**
video-listing-1

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          playlists: [],
+        })
+      );
+    },
+
+    routes() {
+      this.namespace = "api";
+      // auth routes (public)
+      this.post("/auth/signup", signupHandler.bind(this));
+      this.post("/auth/login", loginHandler.bind(this));
```
Comment (vaibhav18matere) [inline src/server.js:80]: why we need to bind every `route` here?

File: src/utils/general.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    : text;
+};
+
+export const delay = (ms) => new Promise((res) => setTimeout(res, ms));
+
+export const formatNumber = (num) => {
+  if (num < 1e3) return num;
+  else if (num >= 1e3 && num < 1e6) return (num / 1e3).toFixed(1) + "K";
+  else if (num >= 1e6 && num < 1e9) return (num / 1e6).toFixed(1) + "M";
+  else if (num >= 1e9 && num < 1e12) return (num / 1e9).toFixed(1) + "B";
```
Comment (vaibhav18matere) [inline src/utils/general.js:13]: what is `1e3` `1e6` and `1e9`? also `K, M, B` ?

**Review by vaibhav18matere — APPROVED** [review]
good work, I just don't understand foll. 2 things. Check those and you are good to go

File: src/server.js (hunk 1 above)
Comment (praveenreddy1798) [reply src/server.js:80]: This is of the mockbee, I haven't made any change

File: src/utils/general.js (hunk 2 above)
Comment (praveenreddy1798) [reply src/utils/general.js:13]: This is to add the necessary suffix for the views count. 10000 => 10k. 1e3 is 10 to the power of 3

**Pull request merged**